File: db4oj/test/src/db4ounit/extensions/fixtures/AbstractFileBasedDb4oFixture.java
/* Copyright (C) 2004 - 2006 db4objects Inc. http://www.db4o.com */

package db4ounit.extensions.fixtures;

import java.io.File;


public abstract class AbstractFileBasedDb4oFixture extends AbstractDb4oFixture {
	
	private final File _yap;

	public AbstractFileBasedDb4oFixture(String fileName) {
		_yap = new File(fileName);
	}

	protected String getAbsolutePath() {
		return _yap.getAbsolutePath();
	}

	public void clean() {
		_yap.delete();
	}


}

File: db4on/Db4oUnit.Extensions/src/Db4oUnit/Extensions/Fixtures/AbstractFileBasedDb4oFixture.cs
namespace Db4oUnit.Extensions.Fixtures
{
	public abstract class AbstractFileBasedDb4oFixture : Db4oUnit.Extensions.Fixtures.AbstractDb4oFixture
	{
		private readonly j4o.io.File _yap;

		public AbstractFileBasedDb4oFixture(Db4oUnit.Extensions.Fixtures.ConfigurationSource
			 configSource, string fileName) : base(configSource)
		{
			_yap = new j4o.io.File(fileName);
		}

		public virtual string GetAbsolutePath()
		{
			return _yap.GetAbsolutePath();
		}

		protected override void DoClean()
		{
			_yap.Delete();
		}
	}
}
